core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -104,12 +107,16 @@ public RecordState validateTransition(RecordState newState) {
                     + "the same as the current state");
             }
 
-            if (this == ACKNOWLEDGED || this == ARCHIVED) {
+            if (this == ARCHIVED) {
```

Comment:
Why do we want to transition records which are already acknowledged, shouldn't we just move the Available ones to Archived when LSO moves?